zipline/finance/commission.py

```diff
@@ -108,3 +111,135 @@ def calculate(self, transaction):
+            option_costs: dictionary specifying per-share commissions in the
+                          option commission tiers defined by IB.
+                          For a description of IB option commission tiers, see
+                          the link below. The dictionary is keyed by a 2d
+                          tuple containing the minimum and maximum option
+                          premium. Values of the dict are the per-contract
+                          commission for an option with premium between these
+                          two minimum and maximum values. For example, the
+                          following dict:
+                                tier_dict = {}
+                                tier_dict[(0.10,  np.inf)] = 0.70
+                                tier_dict[(0.05,    0.10)] = 0.50
+                                tier_dict[(-np.inf, 0.05)] = 0.25
+                          indicates that an option with a premium >= $0.10
+                          has a per-contract commission of $0.70. An option
+                          with a premium between $0.05 and $0.10 has a
+                          commission of $0.50 per-contract, and an option
+                          with a premium less than $0.05 has a per-contract
+                          commission of $0.25.
+            option_min_trade_cost: minimum trade cost for options
+            equity_sids: list of integer sids to apply equity commissions
+            option_sids: list of integer sids to apply option commissions
+
+            https://gdcdyn.interactivebrokers.com/en/index.php?f=commission&p=options1
+        """
+        self.equity_cost = float(equity_cost)
+
+        self.equity_min_trade_cost = None if equity_min_trade_cost is None\
+            else float(equity_min_trade_cost)
+
+        if option_costs is not None:
+            assert isinstance(option_costs, dict)
+        self.option_costs = option_costs
+        self.option_min_trade_cost = None if option_min_trade_cost is None\
+            else float(option_min_trade_cost)
+
+        # check that all sids are integers
+        assert isinstance(equity_sids, list)
+        assert isinstance(option_sids, list)
+        for sid in equity_sids:
+            assert(isinstance(sid, int))
+        for sid in option_sids:
+            assert(isinstance(sid, int))
+
+        # sort for later searching with bisect
+        self.equity_sids = sorted(equity_sids)
+        self.option_sids = sorted(option_sids)
+
+    def __repr__(self):
+        return "{class_name}(cost={cost}, min trade cost={min_trade_cost})"\
+            .format(class_name=self.__class__.__name__,
+                    cost=self.cost,
+                    min_trade_cost=self.min_trade_cost)
+
+    def calculate(self, transaction):
+        """
+        returns a tuple of:
+        (per share commission, total transaction commission)
+        """
+        assert isinstance(transaction.sid, int)
```

Comment:
This assertion is a little misleading because it is an `Asset` which is a subclass of int. It is okay to omit the assertion.